src/plugins/visualize/public/application/components/visualize_editor.tsx

```diff
@@ -0,0 +1,120 @@
+
+import { EMBEDDABLE_ORIGINATING_APP_PARAM } from '../../../../embeddable/public';
+import { removeQueryParam } from '../../../../kibana_utils/public';
+import { useKibana } from '../../../../kibana_react/public';
+import {
+  useChromeVisibility,
+  useSavedVisInstance,
+  useVisualizeAppState,
+  useEditorUpdates,
+  useLinkedSearchUpdates,
+} from '../utils';
+import { VisualizeServices } from '../types';
+import { ExperimentalVisInfo } from './experimental_vis_info';
+import { VisualizeTopNav } from './visualize_top_nav';
+
+export const VisualizeEditor = () => {
+  const { id: visualizationIdFromUrl } = useParams();
+  const [originatingApp, setOriginatingApp] = useState<string>();
+  const { services } = useKibana<VisualizeServices>();
+  const [eventEmitter] = useState(new EventEmitter());
+  const [hasUnsavedChanges, setHasUnsavedChanges] = useState(!visualizationIdFromUrl);
+
+  const isChromeVisible = useChromeVisibility(services.chrome);
+  const { savedVisInstance, visEditorRef, visEditorController } = useSavedVisInstance(
+    services,
+    eventEmitter,
+    isChromeVisible,
+    visualizationIdFromUrl
+  );
+  const { appState, hasUnappliedChanges } = useVisualizeAppState(
+    services,
+    eventEmitter,
+    savedVisInstance
+  );
+  const { isEmbeddableRendered, currentAppState } = useEditorUpdates(
+    services,
+    eventEmitter,
+    setHasUnsavedChanges,
+    isChromeVisible,
+    appState,
+    savedVisInstance,
+    visEditorController
+  );
+  useLinkedSearchUpdates(services, eventEmitter, appState, savedVisInstance);
+
+  useEffect(() => {
+    const originatingAppValue = parse(services.history.location.search)[
+      EMBEDDABLE_ORIGINATING_APP_PARAM
+    ] as string | undefined;
+    if (originatingAppValue) {
+      removeQueryParam(services.history, EMBEDDABLE_ORIGINATING_APP_PARAM);
+    }
+    setOriginatingApp(originatingAppValue);
+  }, [services.history]);
+
+  useEffect(() => {
+    // clean up all registered listeners if any is left
+    return () => {
+      eventEmitter.removeAllListeners();
+    };
+  }, [eventEmitter]);
+
+  return (
+    <div className={`app-container visEditor visEditor--${savedVisInstance?.vis.type.name}`}>
+      {savedVisInstance && appState && currentAppState && (
+        <VisualizeTopNav
+          currentAppState={currentAppState}
+          hasUnsavedChanges={hasUnsavedChanges}
+          setHasUnsavedChanges={setHasUnsavedChanges}
+          isChromeVisible={isChromeVisible}
+          isEmbeddableRendered={isEmbeddableRendered}
+          hasUnappliedChanges={hasUnappliedChanges}
+          originatingApp={originatingApp}
+          savedVisInstance={savedVisInstance}
+          stateContainer={appState}
+          visualizationIdFromUrl={visualizationIdFromUrl}
+        />
+      )}
+      {savedVisInstance?.vis?.type?.isExperimental && <ExperimentalVisInfo />}
+      {savedVisInstance && (
+        <h1 className="euiScreenReaderOnly">
```

Comment:
Probably better to use the EUI Component here as a wrapper instead as I think using the class was an Angular shim
```suggestion
        <EuiScreenReaderOnly>
          <h1>
```